### PR TITLE
Skip coalsce assign in RemoveUnusedPrivatePropertyRector

### LINKED ISSUE #8054: RemoveUnusedPrivatePropertyRector does not recognize access through ??=
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.17.4             |

Rector deletes private properties that are exclusively accessed through `??=`.

## Minimal PHP Code Causing Issue

Since https://getrector.org/demo is down:

```php
class Foo {
    private int $bar;

    public function bar(): int {
        return $this->bar ??= 3;
    }
}
```

## Expected Behaviour

Rector should recognize that `??=` accesses a property.

### PR DESCRIPTION
**Pull request by TomasVotruba**
* Closes https://github.com/rectorphp/rector-src/pull/4490
* Closes https://github.com/rectorphp/rector/issues/8054

### CODE REVIEW

File: packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/AssignedToNodeVisitor.php [hunk 1]
```diff
@@ -22,6 +23,10 @@ public function enterNode(Node $node): ?Node
             return null;
         }
 
+        if ($node instanceof Coalesce) {
+            $node->var->setAttribute(AttributeKey::IS_ASSIGNED_TO, true);
```
Comment (samsonasik) [inline packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/AssignedToNodeVisitor.php:27]: I think common `AssignOp` should be used for it, see `+` usage at https://getrector.com/demo/807f2ae3-e2c3-4810-9b52-96030125a160

that still removed.

Comment (TomasVotruba) [reply packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/AssignedToNodeVisitor.php:27]: Makes sense :+1:  Could you fix that?

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/AssignedToNodeVisitor.php:27]: I will try 👍

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/AssignedToNodeVisitor.php:27]: https://github.com/rectorphp/rector-src/pull/4506

**Pull request merged**